lib/Jsrt/ChakraCore.h

```diff
@@ -163,4 +163,346 @@ JsGetModuleHostInfo(
+/// <param name="value">.</param>
+/// <returns>
+///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+/// </returns>
+CHAKRA_API
+    JsCreateString(
+        const char *content,
+        size_t length,
+        _Out_ JsValueRef *value);
+
+/// <summary>
+///     Create String Variable from Utf8 String
+/// </summary>
+/// <remarks>
+///     <para>
+///     Experimental. We may update the name or behavior until it is stable.
+///     </para>
+/// </remarks>
+/// <param name="content">Pointer to string memory.</param>
+/// <param name="length">Length of the string.</param>
+/// <param name="value">.</param>
+/// <returns>
+///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+/// </returns>
+CHAKRA_API
+    JsCreateStringUtf8(
+        const uint8_t *content,
+        size_t length,
+        _Out_ JsValueRef *value);
+
+/// <summary>
+///     Create String Variable from Utf16 String
+/// </summary>
+/// <remarks>
+///     <para>
+///     Experimental. We may update the name or behavior until it is stable.
+///     </para>
+/// </remarks>
+/// <param name="content">Pointer to string memory.</param>
+/// <param name="length">Length of the string.</param>
+/// <param name="value">.</param>
+/// <returns>
+///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+/// </returns>
+CHAKRA_API
+    JsCreateStringUtf16(
+        const uint16_t *content,
+        size_t length,
+        _Out_ JsValueRef *value);
+
+/// <summary>
+///     Write string value into C string buffer
+/// </summary>
+/// <remarks>
+///     <para>
+///     Experimental. We may update the name or behavior until it is stable.
+///     </para>
+/// </remarks>
+/// <param name="value">JSValueRef</param>
+/// <param name="start">start offset of buffer</param>
+/// <param name="length">length to be written</param>
+/// <param name="buffer">Pointer to buffer.</param>
+/// <param name="written">Total number of characters written</param>
+/// <returns>
+///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+/// </returns>
+CHAKRA_API
+    JsWriteString(
+        JsValueRef value,
+        int start,
+        int length,
+        char* buffer,
+        _Out_opt_ size_t* written);
+
+/// <summary>
+///     Write string value into Utf8 string buffer
+/// </summary>
+/// <remarks>
+///     <para>
+///     Experimental. We may update the name or behavior until it is stable.
+///     </para>
+/// </remarks>
+/// <param name="value">JSValueRef</param>
+/// <param name="buffer">Pointer to buffer.</param>
+/// <param name="bufferSize">Buffer size.</param>
+/// <param name="written">Total number of characters written</param>
+/// <returns>
+///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+/// </returns>
+CHAKRA_API
+    JsWriteStringUtf8(
+        JsValueRef value,
+        uint8_t* buffer,
```

Comment:
Yes, I hate it too, these 3 APIs are indeed inconsistent but required to implement compatible nodecc chakrashim. You can examine their usage in nodecc/xplat. Not sure if we can find a way to make them consistent and satisfy chakrashim.